sqlquerier/sql_querier.go

```diff
@@ -22,21 +22,33 @@ import (
 )
 
 // Query executes a SQL query and returns the number of rows in the result.
-func Query(ctx context.Context, db *sql.DB, query string) (int, error) {
+func Query(ctx context.Context, db *sql.DB, query string) (int, [][]string, error) {
 	if db == nil {
-		return 0, errors.New("no database specified. Please provide one using the --database flag")
+		return 0, nil, errors.New("no database specified. Please provide one using the --database flag")
 	}
 	rows, err := db.Query(query)
 	if err != nil {
-		return 0, err
+		return 0, nil, err
 	}
 	defer rows.Close()
 	n := 0
+
+	// Storing SQL Query results in a string array
+	var result [][]string
```

Comment:
If we only care about the first row+column do we need to store all the other results of the query? From a memory usage standpoint it's cheaper to only store the first cell, and we have some resource constrained places where we're running Localtoast so we'd prefer not to introduce too many additional constraints.

Reply:
Modified the code to store only the first row with all the columns. Since the number of columns depends on the query on the config, it's up to the configuration to select only one row, while the codebase remains as flexible as possible.